modules/apps/analytics/analytics-settings-web/src/main/java/com/liferay/analytics/settings/web/internal/portlet/action/EditWorkspaceConnectionMVCActionCommand.java

```diff
@@ -76,23 +76,39 @@ protected void updateConfigurationProperties(
 
 		if (Objects.equals(cmd, "disconnect")) {
 			_disconnect(actionRequest, configurationProperties);
+
+			return;
 		}
-		else {
-			_connect(actionRequest, configurationProperties);
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long companyId = themeDisplay.getCompanyId();
+
+		boolean upgrade = false;
```

Comment:
Can't tell if this line or 94 is a typo.

Reply:
Line 94 is part of feature flag commit 71bf245. It will be set to true closer to release. Not sure how to properly feature flag it..

Reply:
ah ic